src/main/java/uk/gov/companieshouse/insolvency/data/controller/InsolvencyController.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.companieshouse.insolvency.data.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.delta.Insolvency;
+import uk.gov.companieshouse.insolvency.data.requests.InsolvencyRequest;
+import uk.gov.companieshouse.insolvency.data.service.InsolvencyService;
+
+@RestController
+public class InsolvencyController {
+
+    private InsolvencyService insolvencyService = new InsolvencyService();
+
+    /**
+     * PUT request for insolvency.
+     *
+     * @param  companyNumber  the company number for insolvency
+     * @param  requestBody  the request body containing insolvency data
+     * @return  no response
+     */
+    @PutMapping("/company/{company_number}/insolvency")
+    public ResponseEntity<Void> insolvency(
+            @PathVariable("company_number") String companyNumber,
+            @Validated @RequestBody InsolvencyRequest requestBody
```

Comment:
Do we need to validate?

Reply:
Some of the elements in the insolvency have @Valid or @NotNull, so this ensures its validated, should I remove instead?